### DISCUSSION TITLE
Lesson 2: problem while executing function sFStore

### DISCUSSION THREAD

**Discussion by praxis-dev** [smartcontractkit/full-blockchain-solidity-course-py#1439] [Q&A] [answered]
Hi!

This is the problem as stated in the transactions wimdow:

`transact to StorageFactory.sfStore errored: Error encoding arguments: Error: invalid BigNumber string (argument="value", value="", code=INVALID_ARGUMENT, version=bignumber/5.5.0)`

And here is my code:
```

// SPDX-License_Identifier: MIT

pragma solidity ^0.8.0; 

import "./SimpleStorage.sol";

contract StorageFactory {
    SimpleStorage[] public simpleStorageArray;

    function createSimpleStorageContract() public {
        SimpleStorage simpleStorage = new SimpleStorage();
        simpleStorageArray.push(simpleStorage);
    }

    function sfStore(uint256 _simpleStorageIndex, uint256 _simpleStorageNumber) public {
        // Address
        // ABI
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
        simpleStorage.store(_simpleStorageNumber);
    }
    function sFGet(uint256 _simpleStorageIndex) public view returns (uint256) {
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
        return simpleStorage.retrieve();
    }

}


```

**Comment by LesBLUUs** [comment, marked as answer]
contract StorageFactory is SimpleStorage

try adding the 'is SimpleStorage' after StorageFactory and see if that helps!!

**Reply by cromewar** [reply]
This should help @vaporm give it a try.

**Reply by praxis-dev** [reply]
So now the code looks like this:
```

// SPDX-License_Identifier: MIT

pragma solidity ^0.8.0; 

import "./SimpleStorage.sol";

contract StorageFactory is SimpleStorage {
    SimpleStorage[] public simpleStorageArray;

    function createSimpleStorageContract() public {
        SimpleStorage simpleStorage = new SimpleStorage();
        simpleStorageArray.push(simpleStorage);
    }

    function sfStore(uint256 _simpleStorageIndex, uint256 _simpleStorageNumber) public {
        // Address
        // ABI
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
        simpleStorage.store(_simpleStorageNumber);
    }
    function sFGet(uint256 _simpleStorageIndex) public view returns (uint256) {
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
        return simpleStorage.retrieve();
    }

}

```
And the error persists:

```
transact to StorageFactory.sfStore errored: Error encoding arguments: Error: invalid BigNumber string (argument="value", value="0.55", code=INVALID_ARGUMENT, version=bignumber/5.5.0)
```
Which is exactly the same. Any suggestions?

**Reply by LesBLUUs** [reply]
Hello, and yes I have a suggestion!
try changing this line in your code

 SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
        simpleStorage.store(_simpleStorageNumber);

to this

 SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).store(_simpleStorageNumber);

**Reply by LesBLUUs** [reply]
also the f should be lowercase in your last function perhaps the last function should be adjusted to this as well

```Solidity
 function sfGet(uint256 _simpleStorageIndex) public view returns (uint256) {
        return SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).retrieve();
        
    }

```

**Reply by cromewar** [reply]
Thanks for the help here @LesBLUUs, remember always reference the github repo with the code @vaporm ;)

**Reply by LesBLUUs** [reply]
Anytime!!

**Comment by coodered** [comment]
So have you direct chat to WhatsApp? If is ok I need your WhatsApp number
please

**Reply by mbuv1** [reply]
did this problem get resolved, just started and am stuck at this point:
from solidity:
TypeError: Wrong argument count for function call: 0 arguments given but expected 1.
  --> contracts/StorageFactory.sol:24:16:
   |
24 |         return simpleStorage.retrieve();
   |                ^^^^^^^^^^^^^^^^^^^^^^^^
